server/index.js

```diff
@@ -3,11 +3,207 @@ const db = require('./SCHEMAS/mongodb.js')
+          formatted.answers[answer.answer_id].helpfulness = answer.answer_helpfulness
+          formatted.answers[answer.answer_id].photos = answer.answer_photos
+        }
+      })
+      final.results.push(formatted)
+    })
+    res.send(final)
+  }
+  catch (err) {
+    res.status(500).end(err.message)
+  }
 })
 
+// get answers
+app.get('/qa/questions/:question_id/answers', async (req, res) => {
+  try {
+    let question_id = req.params.question_id
+    let page = req.query.page || 1
+    let count = req.query.page || 5
+
+    let result = await db.answers(question_id)
+
+    let final = {
+      question: result.question_id.toString(),
+      page,
+      count,
+      results: []
+    }
+
+    result.answers.forEach(answer => {
+      if (!answer.answer_reported && count > 0) {
+        let formatted = {
+          answer_id: answer.answer_id,
+          body: answer.answer_body,
+          date: answer.answer_date,
+          aswerer_name: answer.answerer_name,
+          helpfulness: answer.answer_helpfulness,
+          photos: answer.answer_photos
+        }
+
+        final.results.push(formatted)
+        count--
+      }
+    })
+
+    res.send(final)
+  }
+  catch (err) {
+    res.status(500).end(err.message)
+  }
+})
+
+// add a new question
+app.post('/qa/questions', async (req, res) => {
+  try {
+    let body = req.body.body
+    let name = req.body.name
+    let email = req.body.email
+    let product_id = req.body.product_id
+    let counts = await db.countQuery()
+
+    let data = {
+      question_id: counts.questions + 1,
+      product_id,
+      question_body: body,
+      question_date: new Date(),
+      asker_name: name,
+      asker_email: email,
+      question_reported: false,
+      question_helpfulness: 0,
+      answers: []
+    }
+
+    // insert the question and update the count schema
+    await db.insert(data)
+    await db.incrementCount('questions', counts.questions + 1)
+
+    res.send('Created')
+  }
+  catch (err) {
+    res.status(500).end(err.message)
+  }
+})
+
+// add a new answer
+app.post('/qa/questions/:question_id/answers', async (req, res) => {
+  try {
+    let question_id = req.params.question_id
+    let body = req.body.body
+    let name = req.body.name
+    let email = req.body.email
+    let photos = req.body.photos
+    let counts = await db.countQuery()
+
+    let data = {
+      id: counts.answers + 1,
+      body: body,
+      date_written: new Date(),
+      answerer_name: name,
+      answerer_email: email,
+      reported: false,
+      helpful: 0,
+      answer_photos: photos
+    }
+
+    // insert answer and increment the counter schema
+    await db.answerInsert(question_id, data)
+    await db.incrementCount('answers', counts.answers + 1)
+
+    res.send('Created')
+  }
+  catch (err) {
+    res.status(500).end(err.message)
+  }
+})
+
+// update helpfulness of question
+app.put('/qa/questions/:question_id/helpful', async (req, res) => {
```

Comment:
I know there are some small differences between POST and PUT, and I'd be curious to hear your reasons for having some of your routs be PUT requests. Just a note for future discussion.

Reply:
This was documented in the original documentation. The helpful updates and reported updates were all PUT requests according to the documentation. I'd have to look at why, but my assumption is because a PUT request will update something that already exits, whereas a POST request is adding something new.